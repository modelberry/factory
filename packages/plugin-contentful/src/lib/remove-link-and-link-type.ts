import { SysLink } from 'contentful-management/types'

export interface RemoveLinkAndLinkType {
  fieldValue: SysLink | SysLink[] | Record<string, any>
}

export const removeLinkAndLinkType = ({
  fieldValue,
}: RemoveLinkAndLinkType) => {
  if (!fieldValue) return fieldValue
  if (Array.isArray(fieldValue)) {
    fieldValue.forEach((field) => {
      if (field.sys) {
        delete (field.sys as any).type
        delete (field.sys as any).linkType
      }
    })
  } else {
    if (fieldValue.sys) {
      delete (fieldValue.sys as any).type
      delete (fieldValue.sys as any).linkType
    }
  }
  return fieldValue
}
