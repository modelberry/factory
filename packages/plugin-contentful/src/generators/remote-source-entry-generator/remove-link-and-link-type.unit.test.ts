import { removeLinkAndLinkType } from './remove-link-and-link-type'

describe('removeLinkAndLinkType should', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('remove link and linkType from items', async () => {
    const fieldValue = {
      id: 'footer',
      name: 'Footer',
      type: 'Array',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
      items: {
        type: 'Link',
        validations: [{ linkContentType: ['navigationSegment'] }],
        linkType: 'Entry',
      },
    }
    const remoteEntryFieldValue = removeLinkAndLinkType({
      fieldValue,
    })
    expect(remoteEntryFieldValue).toMatchSnapshot()
  })
  test('remove link and linkType', async () => {
    const fieldValue = {
      id: 'social',
      name: 'Social',
      type: 'Link',
      localized: false,
      required: false,
      validations: [{ linkContentType: ['navigationSegment'] }],
      disabled: false,
      omitted: false,
      linkType: 'Entry',
    }
    const remoteEntryFieldValue = removeLinkAndLinkType({
      fieldValue,
    })
    expect(remoteEntryFieldValue).toMatchSnapshot()
  })
})
