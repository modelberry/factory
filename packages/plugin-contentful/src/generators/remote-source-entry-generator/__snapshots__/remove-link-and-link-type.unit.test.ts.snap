// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeLinkAndLinkType should remove link and linkType 1`] = `
{
  "disabled": false,
  "id": "social",
  "linkType": "Entry",
  "localized": false,
  "name": "Social",
  "omitted": false,
  "required": false,
  "type": "Link",
  "validations": [
    {
      "linkContentType": [
        "navigationSegment",
      ],
    },
  ],
}
`;

exports[`removeLinkAndLinkType should remove link and linkType from items 1`] = `
{
  "disabled": false,
  "id": "footer",
  "items": {
    "linkType": "Entry",
    "type": "Link",
    "validations": [
      {
        "linkContentType": [
          "navigationSegment",
        ],
      },
    ],
  },
  "localized": false,
  "name": "Footer",
  "omitted": false,
  "required": false,
  "type": "Array",
  "validations": [],
}
`;
